selfdrive/car/subaru/carcontroller.py

```diff
@@ -32,13 +37,31 @@ def update(self, CC, CS, now_nanos):
       new_steer = int(round(apply_steer))
       apply_steer = apply_driver_steer_torque_limits(new_steer, self.apply_steer_last, CS.out.steeringTorque, self.p)
 
+      apply_steer_req = 1
+
       if not CC.latActive:
         apply_steer = 0
+        apply_steer_req = 0
+
+      if self.CP.carFingerprint in STEER_LIMITED:
+        # Count up to MAX_STEER_RATE_FRAMES, at which point we need to cut torque to avoid a steering fault
+        if abs(CS.out.steeringRateDeg) >= MAX_STEER_RATE:
+          self.steer_rate_counter += 1
+        else:
+          self.steer_rate_counter = 0
+        
+        if self.steer_rate_counter > MAX_STEER_RATE_FRAMES:
+          apply_steer_req = 0
+          self.steer_rate_counter = 0
+
+        # Any steering past 90 appears to cause temp fault
+        if abs(CS.out.steeringAngleDeg) > MAX_STEER_ANGLE:
+          apply_steer_req = 0
```

Comment:
we have a few safety modes which use this same logic, might be good to abstract it in another PR

Reply:
made one here: https://github.com/commaai/openpilot/pull/29140